diff --git a/deps/libgo2/src/display.c b/deps/libgo2/src/display.c
index b91cf0624b..689198333a 100644
--- a/deps/libgo2/src/display.c
+++ b/deps/libgo2/src/display.c
@@ -593,7 +593,7 @@ void go2_surface_blit(go2_surface_t* srcSurface, int srcX, int srcY, int srcWidt
     switch (rotation)
     {
         case GO2_ROTATION_DEGREES_0:
-            src.rotation = 0;
+            src.rotation = HAL_TRANSFORM_ROT_90;
             break;
 
         case GO2_ROTATION_DEGREES_90:
@@ -601,11 +601,11 @@ void go2_surface_blit(go2_surface_t* srcSurface, int srcX, int srcY, int srcWidt
             break;
 
         case GO2_ROTATION_DEGREES_180:
-            src.rotation = HAL_TRANSFORM_ROT_180;
+            src.rotation = HAL_TRANSFORM_ROT_90;
             break;
 
         case GO2_ROTATION_DEGREES_270:
-            src.rotation = HAL_TRANSFORM_ROT_270;
+            src.rotation = HAL_TRANSFORM_ROT_90;
             break;
 
         default:
diff --git a/gfx/drivers/oga_gfx.c b/gfx/drivers/oga_gfx.c
index bb71b27159..42b2934ad8 100644
--- a/gfx/drivers/oga_gfx.c
+++ b/gfx/drivers/oga_gfx.c
@@ -708,16 +708,16 @@ static void oga_set_rotation(void *data, unsigned rotation)
    switch (rotation)
    {
    case 0:
-      vid->rotation = HAL_TRANSFORM_ROT_270;
+      vid->rotation = HAL_TRANSFORM_ROT_90;
       break;
    case 1:
-      vid->rotation = HAL_TRANSFORM_ROT_180;
+      vid->rotation = HAL_TRANSFORM_ROT_90;
       break;
    case 2:
       vid->rotation = HAL_TRANSFORM_ROT_90;
       break;
    case 3:
-      vid->rotation = 0;
+      vid->rotation = HAL_TRANSFORM_ROT_90;
       break;
    default:
       RARCH_ERR("Unhandled rotation %hu\n", rotation);
